version: '3'

services:
  # Nginx
  nginx:
    container_name: nginx
    image: nginx:latest
    ports:
      - '80:80'
    volumes:
      - ./conf/nginx.conf:/etc/nginx/nginx.conf
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 3

  # БД для бэкофиса.
  backoffice-db:
    container_name: postgres
    image: postgres:13.1
    networks:
      - gergoltz_test
    volumes:
      - /data/gergoltz_test/backoffice/postgres/data:/var/lib/postgresql/data
    environment:
      - POSTGRES_USER=${BACKOFFICE_DB_USER}
      - POSTGRES_PASSWORD=${BACKOFFICE_DB_PASSWORD}
      - POSTGRES_DB=${BACKOFFICE_DB_NAME}
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 3

  # Бэкофис.
  backoffice-app:
    container_name: django_app
    build:
      context: backoffice
      dockerfile: Dockerfile
    networks:
      - gergoltz_test
    depends_on:
      - backoffice-db
      - nginx
    ports:
      - 8000:8000
    volumes:
      - ./backoffice:/app
    command: >
      bash -c "
      pip install shapely
      && python manage.py test > test_results.txt
      && python manage.py migrate
      && python manage.py loaddata test_data.json
      && python manage.py runserver 0.0.0.0:8000"
    environment:
      - BACKOFFICE_DB_NAME=${BACKOFFICE_DB_NAME}
      - BACKOFFICE_USER_DB=${BACKOFFICE_DB_USER}
      - BACKOFFICE_PASSWORD_DB=${BACKOFFICE_DB_PASSWORD}
      - BACKOFFICE_HOST_DB=${BACKOFFICE_DB_HOST}
      - BACKOFFICE_PORT_DB=${BACKOFFICE_DB_PORT}
      - DEBUG=True
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 3

networks:
  gergoltz_test: